### PR TITLE
New 404 page

### LINKED ISSUE #256: New 404 Page
## Problem

Our current 404 page is unhelpful to users and unattractive. We should implement a new one with a built-in search bar and styling that reinforces Openverse's brand. 

## Description

**Edit: the page is designed and ready for work!**

Figma Link: https://www.figma.com/file/w60dl1XPUvSaRncv1Utmnb/Openverse-Releases?node-id=133%3A6869

image

## Original brief

> 
> We should design a new page (cc @panchovm) that provides better functionality and aesthetics. Some ideas for functionality:
> 
> - A center search bar to encourage a user to submit a query
> - More helpful information (for example, if they 404 on a `/photo/{id}` url, say something like "The photo you tried to view may have been removed" etc.)
> - Perhaps some featured collections or a random piece of media

### PR DESCRIPTION
**Pull request by zackkrida**
## Fixes

Fixes #256 by @zackkrida 

## Description

Implements the new 404 page, with a search bar for an 'all results' search. Sadly this needs to be based on the VHeader branch since there's searchbar functionality fixes there.

CleanShot 2022-01-06 at 15 25 01@2x

CleanShot 2022-01-06 at 15 24 47@2x

## Testing Instructions

Visit any fake url, like http://localhost:8443/dfsgfefad, and do some testing:

- Observe the page on different viewport sizes
- Submit a search
- Try the 'home' link

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by panchovm** [comment]
I noticed some errors in the mockups that I forgot to update. Sorry for the mistake. The desktop and mobile versions are now ready with the following changes:

1. Logo size is smaller than the previous one on desktop.
2. Text title changed to be "The content you’re looking for seems to have disappeared."
3. On desktop, the title is `Heading 1`.
4. On mobile, the title is `Heading 4`.
5. Search bar border is `dark-charcoal`.
6. In the Search bar, placeholder text and search icon colors are `dark-charcoal-70`.

**Comment by zackkrida** [comment]
@panchovm I've implemented your suggestions and you can re-review!

**Comment by panchovm** [comment]
I could not identify the color used for the placeholder text, but likely it is correct. We are using the `dark-charcoal-70` shade.

Regarding the description below title, I like the "search for content with the field below" but am unsure about the "Return to" as for some users this link might be the first one visiting openverse. That being said, it makes sense to me to use "Go to" instead.

On the other hand, the current search bar design (input + button) is inconsistent across desktop and mobile. This is a design mistake that I was considering tackling once working on adding the "clear action" within the input.

Despite the text doubt, everything else looks great to me.

File: src/components/VInputField/VInputField.vue [hunk 1]
```diff
@@ -88,4 +88,8 @@ export default {
 .input-field:focus-within .info {
   @apply text-dark-charcoal;
 }
+
```
Comment (krysal) [inline src/components/VInputField/VInputField.vue:91]: The `placeholder-dark-charcoal-70` class in this component should be replaced by `placeholder:text-dark-charcoal-70`.

On the other hand, the style below can be replaced by the `placeholder:font-normal` utility too.

File: src/layouts/error.vue [hunk 2]
```diff
@@ -1,30 +1,116 @@
 <template>
-  <div class="grid-container full">
-    <main class="not-found">
-      <h1 v-if="error.statusCode === 404">
-        {{ $t('error.not-found') }}
+  <main class="bg-yellow h-screen relative page-404">
+    <Logo class="pt-6 lg:pt-8 ms-6 lg:ms-10 w-30 h-auto" />
```
Comment (krysal) [inline src/layouts/error.vue:3]: Shouldn't the logo be a link to home?

**Review by krysal — COMMENTED** [review]
Looks great! It just needs the adjustments on the placeholder text that Francisco mentioned.

File: src/locales/en.json [hunk 3]
```diff
@@ -1,4 +1,10 @@
 {
+  "404": {
+    "title": "The content you’re looking for seems to have disappeared.",
+    "main": "Return to {link} or search for content with the field below.",
```
Comment (krysal) [inline src/locales/en.json:4]: The text changed in the mockup:
```suggestion
    "main": "Go to {link} or search for something similar from the field below.",
```

File: src/layouts/error.vue (hunk 2 above)
Comment (obulat) [reply src/layouts/error.vue:3]: And it needs something for a11y, or maybe aria-hidden if it's not a link?

**Review by obulat — APPROVED** [review]
Looks great!

File: src/locales/en.json (hunk 3 above)
Comment (zackkrida) [reply src/locales/en.json:4]: @krysal @panchovm I'm not sure about this text. The `{link}` goes to the home page, so 'return to' makes sense to me, and 'search for something similar' doesn't make a lot of sense to me. Similar to what?

**Comment by zackkrida** [comment]
The logo is now a link with screen reader text, thank you for the suggestion!

**Comment by zackkrida** [comment]
> I noticed a horizontal scroll when testing it through the responsive design mode in firefox.

I was not aboe to recreate this @panchovm but I did add some extra styling to prevent it.

**Comment by obulat** [comment]
> I noticed a horizontal scroll when testing it through the responsive design mode in firefox.
> 
> 

These scrollbars look like they come from the Firefox window, not Openverse. Could you try making the firefox responsive design mode window larger, or decreasing the scale, maybe, @panchovm?

File: src/locales/en.json (hunk 3 above)
Comment (krysal) [reply src/locales/en.json:4]: If someone comes from another website to this page the "Go to ..." seems more appropriate, the "Return" makes me think of the go back feature of the browser, but maybe that is just me.

About _something similar_, I was assuming an image or audio that it's no longer shown for been taken down or whatever reason, but none of these are big deals neither.

**Review by krysal — APPROVED** [review]
Great!

**Comment by panchovm** [comment]
I resized the window and tried different orientations @obulat and now it works perfectly.

**Review by panchovm — APPROVED** [review]
It looks and works great.

File: src/locales/en.json (hunk 3 above)
Comment (zackkrida) [reply src/locales/en.json:4]: Both of those explanations make sense, thank you!

**Pull request merged**